homeassistant/components/unifi/sensor.py

```diff
@@ -110,11 +112,22 @@ def async_wlan_client_value_fn(controller: UniFiController, wlan: Wlan) -> int:
 @callback
 def async_device_uptime_value_fn(
     controller: UniFiController, device: Device
-) -> datetime:
-    """Calculate the uptime of the device."""
-    return (dt_util.now() - timedelta(seconds=device.uptime)).replace(
-        second=0, microsecond=0
-    )
+) -> datetime | None:
+    """Calculate the approximate time the device started (based on uptime returned from API, in seconds)."""
+    if device.uptime <= 0:
+        # Library defaults to 0 if uptime is not provided, e.g. when offline
```

Comment:
Should I change the library to report None rather than 0?

Reply:
I think that would make sense, at least for other potential consumers. Would probably need to tag a new major version because of the BC break though

Reply:
Probably no other primary uses. The library is built for the integration